imap/ctl_mboxlist.c

```diff
@@ -1007,6 +1007,9 @@ int main(int argc, char *argv[])
 
     case M_POPULATE:
         syslog(LOG_NOTICE, "%spopulating mupdate", warn_only ? "test " : "");
+#if defined __GNUC__ &&  __GNUC__ > 6
```

Comment:
Ironically, all the #ifdef noise makes it less clear that this is a falling through...

I would maybe put something like this into lib/util.h:

```
#if defined __GNUC__ &&  __GNUC__ > 6
    #define GCC_FALLTHROUGH __attribute__((fallthrough));
#else
    #define GCC_FALLTHROUGH /* fall through */
#endif
```

and then the switches can do something like:

```
case M_POPULATE:
    syslog(blah);
    GCC_FALLTHROUGH

case DUMP:
    ...
```

An even better place than lib/util.c would be if it wound up in config.h, since everything includes that.  I don't remember off the top of my head how stuff gets into there (maybe config.h.in?) but I'm sure it's not hard to work out